web/src/message_events.js

```diff
@@ -720,30 +712,7 @@ export function update_messages(events) {
         }
     }
 
-    // If a topic was edited, we re-render the whole view to get any
-    // propagated edits to be updated (since the topic edits can have
-    // changed the correct grouping of messages).
-    if (any_topic_edited || any_stream_changed) {
-        // However, we don't need to rerender message_list if
-        // we just changed the narrow earlier in this function.
-        //
-        // TODO: We can potentially optimize this logic to avoid
-        // calling `update_muting_and_rerender` if the muted
-        // messages would not match the view before or after this
-        // edit.  Doing so could save significant work, since most
-        // topic edits will not match the current topic narrow in
-        // large organizations.
```

Comment:
So the potential bug that I'm confused about here is that this comment seems to think it's important to call `update_muting_and_rerender` when moving topics, and I believe that -- it should be necessary to do something if we, say, move a message that matches the filter from a normal topic into a muted topic if the view is one that hides muted topics. 

But reading the code here, it doesn't appear that the existing logic can possibly have done that; so this is probably just an existing bug that we need to fix. So I'll open a separate issue for it.

Reply:
Opened #31977 for this.